src/Cognito.re

```diff
@@ -65,10 +65,11 @@ module Client = {
   // returns an error code (4XX, 5XX), that should be handled elsewhere.
   //
   // TODO: More generic implemntation (no Js.* types)
-  let request = (config, operation, params: Js.Dict.t(Js.Json.t)) => {
+  let request = (config, operation: operation, params: Js.Dict.t(Js.Json.t)) => {
```

Comment:
I'd delete this and add this type to the function signature in the module signature above:

```reason
  let request:
    (t, string, Js.Dict.t(Js.Json.t)) =>
    (t, operation, Js.Dict.t(Js.Json.t)) =>
    Future.t(Belt.Result.t(response, [< error]));
    Future.t(Belt.Result.t(response, [< error]));
```

Reply:
you might have to add `type operation` there too.

Reply:
Done. Compiler picks up `Types.operation` so did not have to annotate.